controllers/jenkins/pipelinerun/pipelinerun_controller.go

```diff
@@ -309,6 +323,24 @@ func (r *Reconciler) makePipelineRunOrphan(ctx context.Context, pr *v1alpha3.Pip
 	return r.updateStatus(ctx, &pr.Status, client.ObjectKey{Namespace: pr.Namespace, Name: pr.Name})
 }
 
+func (r *Reconciler) getOrCreateJenkinsCoreIfHasCreator(annotations map[string]string) (*core.JenkinsCore, error) {
```

Comment:
No matter if there is a creator or not, this function always returns an object. So, I suggest changing the function name to `getOrCreateJenkinsCore`